registrations/views.py

```diff
@@ -213,3 +213,23 @@ def post(self, request, *args, **kwargs):
             status = 400
 
         return Response(resp, status=status)
+
+
+class PersonnelCodeView(APIView):
+
+    """ PersonnelCodeView Interaction
+        GET - returns a list of identities with personnel codes
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, *args, **kwargs):
+        status = 200
+        resp = {"results": []}
+
+        identities = utils.search_identities(
+            {"details__has_key": "personnel_code"})
+
+        for identity in identities:
+            resp["results"].append(identity['details'].get('personnel_code'))
+
+        return Response(resp, status=status)
```

Comment:
What are we trying to achieve with this view?

If the goal is to have an endpoint to check whether a clinic code is valid or not, then maybe we should do that, instead of returning all valid clinic codes.

If the client of this endpoint is going to use all of the clinic codes, then this endpoint is fine.

Reply:
The goal of the vas2nets line is to work even when mainone is down, so they would request this periodically and save it to validate against. If mainone is down then they validate with what they have saved.